cmd/node_agent/na/nodeagent.go

```diff
@@ -78,43 +79,98 @@ type platformSpecificRequest interface {
 	IsProperPlatform() bool
 }
 
+// CAGrpcClient is for implementing the GRPC client to talk to CA.
+type CAGrpcClient interface {
```

Comment:
Sorry I was wrong. The generated one has the following signature: `HandleCSR(ctx context.Context, in *Request, opts ...grpc.CallOption) (*Response, error)`. Guess we need a wrapper here since we're not passing in ctx and opts.